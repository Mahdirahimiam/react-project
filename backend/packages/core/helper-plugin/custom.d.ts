export {};

declare global {
  interface Window {
    strapi: {
      backendURL: string;
      telemetryDisabled: boolean;
      projectType: 'Enterprise' | 'Community';
    };
  }
}

import { StrapiTheme } from '@strapi/design-system';

declare module 'styled-components' {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  export interface DefaultTheme extends StrapiTheme {}
}
