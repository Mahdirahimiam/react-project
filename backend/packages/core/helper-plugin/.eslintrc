{
  "root": true,
  "extends": ["custom/front/typescript", "plugin:storybook/recommended"],
  "rules": {
    "import/no-unresolved": [
      "error",
      {
        // Eslint does not detect the package.json exports field
        // and reports broken imports for the following imports
        "ignore": ["@strapi/design-system/v2"]
      }
    ]
  },
  "overrides": [
    {
      "files": ["./jest.config.front.js", "./webpack.config.js"],
      "env": {
        "node": true
      },
      "rules": {
        "@typescript-eslint/no-var-requires": "off"
      }
    },
    {
      "files": ["./tests/*"],
      "env": {
        "jest": true
      }
    }
  ]
}
