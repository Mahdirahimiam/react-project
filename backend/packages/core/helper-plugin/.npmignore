############################
# dts
############################

dist/**/*.d.ts.map