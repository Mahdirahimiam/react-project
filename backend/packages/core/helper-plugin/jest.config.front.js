module.exports = {
  preset: '../../../jest-preset.front.js',
  collectCoverageFrom: ['<rootDir>/packages/core/helper-plugin/src/**/*.js'],
  displayName: 'Helper plugin',
  moduleNameMapper: {
    '^@tests/(.*)$': '<rootDir>/tests/$1',
  },
  setupFilesAfterEnv: ['./tests/setup.ts'],
};
