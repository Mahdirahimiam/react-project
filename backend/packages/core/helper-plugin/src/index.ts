/* -------------------------------------------------------------------------------------------------
 * Components
 * -----------------------------------------------------------------------------------------------*/

export * from './components/AnErrorOccurred';
export * from './components/CheckPagePermissions';
export * from './components/CheckPermissions';
export * from './components/ConfirmDialog';
export * from './components/ContentBox';
export * from './components/DateTimePicker';
export * from './components/DynamicTable';
export * from './components/Table';
export * from './components/EmptyStateLayout';
export * from './components/EmptyBodyTable';
export * from './components/FilterListURLQuery';
export * from './components/FilterPopoverURLQuery';
export * from './components/Form';
export * from './components/GenericInput';
export * from './components/InjectionZone';
export * from './components/Link';
export * from './components/LinkButton';
export * from './components/LoadingIndicatorPage';
export * from './components/NoContent';
export * from './components/NoMedia';
export * from './components/NoPermissions';
export * from './components/NotAllowedInput';
export * from './components/PageSizeURLQuery';
export * from './components/PaginationURLQuery';
export * from './components/ReactSelect';
export * from './components/RelativeTime';
export * from './components/SearchURLQuery';
export * from './components/SettingsPageTitle';
export * from './components/Status';

/* -------------------------------------------------------------------------------------------------
 * Content Manager
 * -----------------------------------------------------------------------------------------------*/

export {
  useCMEditViewDataManager,
  ContentManagerEditViewDataManagerContext,
  type CMEditViewDataManagerContextValue,
  type ContentType,
} from './content-manager/CMEditViewDataManager';
export * from './content-manager/utils/getAttributeInfos';

/* -------------------------------------------------------------------------------------------------
 * Features
 * -----------------------------------------------------------------------------------------------*/

export * from './features/AppInfo';
export * from './features/AutoReloadOverlayBlocker';
export * from './features/CustomFields';
export * from './features/GuidedTour';
export * from './features/Library';
export * from './features/Notifications';
export * from './features/OverlayBlocker';
export * from './features/RBAC';
export * from './features/StrapiApp';
export * from './features/Tracking';

/* -------------------------------------------------------------------------------------------------
 * Hooks
 * -----------------------------------------------------------------------------------------------*/

export * from './hooks/useAPIErrorHandler';
export * from './hooks/useCallbackRef';
export * from './hooks/useClipboard';
export * from './hooks/useCollator';
export * from './hooks/useFetchClient';
export * from './hooks/useFilter';
export * from './hooks/useFocusInputField';
export * from './hooks/useFocusWhenNavigate';
export * from './hooks/useLockScroll';
export * from './hooks/usePersistentState';
export * from './hooks/useQuery';
export * from './hooks/useQueryParams';
export * from './hooks/useRBAC';
export * from './hooks/useSelectionState';

/* -------------------------------------------------------------------------------------------------
 * Icons
 * -----------------------------------------------------------------------------------------------*/

export * from './icons/RemoveRoundedButton';
export * from './icons/SortIcon';

/* -------------------------------------------------------------------------------------------------
 * Utils
 * -----------------------------------------------------------------------------------------------*/

export * from './content-manager/utils/contentManagementUtilRemoveFieldsFromData';
export * from './content-manager/utils/formatContentTypeData';
export * from './utils/auth';
export * from './utils/awaitToJs';
export * from './utils/difference';
export * from './utils/getAPIInnerErrors';
export * from './utils/getFetchClient';
export * from './utils/getFileExtension';
export * from './utils/getYupInnerErrors';
export * from './utils/hasPermissions';
export * from './utils/normalizeAPIError';
export * from './utils/prefixFileUrlWithBackendUrl';
export * from './utils/prefixPluginTranslations';
export * from './utils/pxToRem';
export * from './utils/request';
export * from './utils/setHexOpacity';
export * from './utils/stopPropagation';
export * from './utils/translatedErrors';
export * from './utils/wrapAxiosInstance';

/* -------------------------------------------------------------------------------------------------
 * Types
 * -----------------------------------------------------------------------------------------------*/

export type { TranslationMessage, FilterData } from './types';
