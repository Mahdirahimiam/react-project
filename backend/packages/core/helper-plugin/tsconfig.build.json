{
  "extends": "./tsconfig.json",
  "exclude": ["**/*.test.tsx", "**/*.test.ts"],
  "compilerOptions": {
    "declarationMap": true,
    "outDir": "./dist"
  }
}
