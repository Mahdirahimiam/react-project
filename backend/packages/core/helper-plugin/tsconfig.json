{
  "extends": "tsconfig/client.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@tests/*": ["./tests/*"]
    }
  },
  "include": ["src", "custom.d.ts"]
}
