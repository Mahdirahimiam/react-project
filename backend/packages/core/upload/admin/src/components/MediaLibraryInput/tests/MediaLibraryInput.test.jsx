import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { NotificationsProvider } from '@strapi/helper-plugin';
import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import { MediaLibraryInput } from '..';
import en from '../../../translations/en.json';

jest.mock('../../../utils/downloadFile');

jest.mock('../../../utils', () => ({
  ...jest.requireActual('../../../utils'),
  getTrad: (x) => x,
}));

jest.mock('react-intl', () => ({
  FormattedMessage: ({ id }) => id,
  useIntl: () => ({ formatMessage: jest.fn(({ id }) => en[id] || id) }),
}));
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },
});

const renderCompo = (
  props = {
    onChange: jest.fn(),
    name: 'test',
    intlLabel: {
      id: 'default',
      defaultMessage: 'default message',
    },
  }
) =>
  render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <NotificationsProvider toggleNotification={jest.fn()}>
          <MediaLibraryInput {...props} />
        </NotificationsProvider>
      </ThemeProvider>
    </QueryClientProvider>,
    { container: document.getElementById('app') }
  );

describe('<MediaLibraryInput />', () => {
  it('renders and matches the snapshot', () => {
    renderCompo();

    expect(document.body).toMatchSnapshot();
  });
});
