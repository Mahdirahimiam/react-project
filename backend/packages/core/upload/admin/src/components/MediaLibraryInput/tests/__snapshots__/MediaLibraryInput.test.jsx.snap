// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaLibraryInput /> renders and matches the snapshot 1`] = `
.c19 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c3 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c17 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #666687;
}

.c0 {
  margin-left: -250px;
  position: fixed;
  left: 50%;
  top: 2.875rem;
  z-index: 10;
  width: 31.25rem;
}

.c5 {
  background: #f6f6f9;
  padding: 8px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
}

.c6 {
  display: grid;
  position: relative;
}

.c8 {
  padding-right: 8px;
  padding-left: 8px;
  overflow: hidden;
  width: 100%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 124px;
}

.c13 {
  width: 100%;
  height: 100%;
}

.c15 {
  color: #4945ff;
  margin-bottom: 12px;
  width: 30px;
  height: 24px;
}

.c1 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
}

.c2 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c16 path {
  fill: #4945ff;
}

.c7 {
  grid-template-columns: auto 1fr auto;
  grid-template-areas: 'startAction slides endAction';
}

.c10 {
  grid-area: slides;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<body>
  <div>
    <div
      class="c0 c1"
    />
    <div
      class=""
    >
      <div
        class="c2"
      >
        <label
          class="c3 c4"
          for=":r0:"
        >
          default
        </label>
        <div
          class=""
          id=":r0:"
        >
          <div
            class="c5"
          >
            <section
              aria-label="default"
              aria-roledescription="carousel"
              class="c6 c7"
              display="grid"
            >
              <div
                aria-live="polite"
                class="c8 c9 c10"
                overflow="hidden"
              >
                <div
                  aria-label="upload.mediaLibraryInput.slideCount"
                  aria-roledescription="slide"
                  class="c11 c12"
                  display="flex"
                  role="group"
                >
                  <button
                    class="c13 c14"
                    style="cursor: pointer;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="c15 c16"
                      fill="none"
                      height="1rem"
                      viewBox="0 0 24 20"
                      width="1rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21.569 2.398H7.829v1.586h13.74c.47 0 .826.5.826 1.094v9.853l-2.791-3.17a2.13 2.13 0 0 0-.74-.55 2.214 2.214 0 0 0-.912-.196 2.215 2.215 0 0 0-.912.191 2.131 2.131 0 0 0-.74.546l-2.93 3.385-2.973-3.36a2.147 2.147 0 0 0-.741-.545 2.228 2.228 0 0 0-1.824.007c-.286.13-.538.319-.739.553l-2.931 3.432V7.653H2.51v9.894c.023.153.06.304.108.452v.127l.041.095c.057.142.126.28.207.412l.099.15c.074.107.157.207.247.302l.124.119c.13.118.275.222.43.309h.024c.36.214.775.327 1.198.325h16.515c.36-.004.716-.085 1.039-.24.323-.153.606-.375.827-.648a2.78 2.78 0 0 0 .504-.888c.066-.217.108-.44.124-.666V5.078a2.497 2.497 0 0 0-.652-1.81 2.706 2.706 0 0 0-1.776-.87Z"
                        fill="#32324D"
                      />
                      <path
                        d="M12.552 9.199c.912 0 1.651-.71 1.651-1.585 0-.876-.74-1.586-1.651-1.586-.912 0-1.652.71-1.652 1.586 0 .875.74 1.585 1.652 1.585ZM3.303 6.408h.826V3.997h2.477V2.41H4.129V0h-.826c-.219 0-.85.002-.826 0v2.411H0v1.586h2.477v2.41h.826Z"
                        fill="#32324D"
                      />
                    </svg>
                    <span
                      class="c17 c18"
                      style="text-align: center;"
                    >
                      upload.mediaLibraryInput.placeholder
                    </span>
                  </button>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c19"
    >
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-log"
        role="log"
      />
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-status"
        role="status"
      />
      <p
        aria-live="assertive"
        aria-relevant="all"
        id="live-region-alert"
        role="alert"
      />
    </div>
  </div>
</body>
`;
